### PR TITLE
Create AddClaimCommandParser object

### PR DESCRIPTION
**Pull request by RezwanAhmed123**
Created a parser for AddClaimCommand such that the user inputs can
be successfully parsed.

This class is still a placeholder and unable to check for valid
claim ids and insurance plan ownership.

### CODE REVIEW

File: src/main/java/seedu/address/logic/Messages.java [hunk 1]
```diff
@@ -50,4 +50,13 @@ public static String format(Person person) {
         return builder.toString();
     }
 
+    /**
+     * Formats the {@code ClaimAmount} for display to the user by converting from cents to dollar and cents.
+     * The formatting was generated using ChatGPT.
+     */
+    public static String formatClaimAmount(int claimAmount) {
+        int centsInADollar = 100;
+        return String.format("$%d.%02d", claimAmount / centsInADollar, claimAmount % centsInADollar);
```
Comment (matthewyip1511) [inline src/main/java/seedu/address/logic/Messages.java:59]: Maybe we can abstract out `claimAmount / centsInADollar` and `claimAmount % centsInADollar` here for easier reading? We could do

```
int claimAmountDollarsPart = claimAmount / centsInADollar;
int claimAmountCentsPart = claimAmount % centsInADollar;
```

File: src/main/java/seedu/address/logic/commands/AddClaimCommand.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ */
+public class AddClaimCommand extends Command {
+    public static final String COMMAND_WORD = "addClaim";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Adds an open claim to the insurance plan of the client identified "
+            + "by their client id. \n"
+            + "Parameters: INDEX (must be a positive integer), "
+            + " INSURANCE_PLAN_ID (must be a valid ID), "
+            + " Claim_ID (must be a valid ID), "
+            + " Claim amount (must be a valid integer, convert from dollars and cents to just cents) \n"
```
Comment (matthewyip1511) [inline src/main/java/seedu/address/logic/commands/AddClaimCommand.java:27]: what about `expressed in cents` for greater simplicity 🙂

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -129,6 +129,37 @@ public static int parseInsurancePlan(String insurancePlanId) throws ParseExcepti
...
+     *
+     * @throws ParseException if the claim amount is not a valid positive integer
+     */
+    public static int parseClaimAmount(String claimAmount) throws ParseException {
+        requireNonNull(claimAmount);
+        int claimAmountInt;
+        try {
+            claimAmountInt = Integer.parseInt(claimAmount.trim());
+            if (claimAmountInt < 0) {
+                throw new ParseException("Unplayable, Claim Amount cannot be negative");
```
Comment (matthewyip1511) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:155]: Did you mean not payable 😅

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -129,6 +129,37 @@ public static int parseInsurancePlan(String insurancePlanId) throws ParseExcepti
...
+            }
+        } catch (NumberFormatException e) {
+            throw new ParseException("Unplayable, claim Amount must be a positive integer");
```
Comment (matthewyip1511) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:158]: Likewise as above

**Review by matthewyip1511 — CHANGES_REQUESTED** [review]
Overall LGTM, other than some small changes 🙂

File: src/main/java/seedu/address/logic/commands/AddClaimCommand.java (hunk 2 above)
Comment (zi-yii) [inline src/main/java/seedu/address/logic/commands/AddClaimCommand.java:27]: Should claim amount be in dollars and cents when the user inputs it, while we save claim amount as cents in the system? The display of claim amount on the GUI to also be in dollars and cents for easier understanding. This way the app can be more user friendly. What do you think?

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 4 above)
Comment (zi-yii) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:158]: For lines 155 and 158, to standardise the capitalisation of claim amount? Maybe just keep claim amount as all small letters?

**Review by zi-yii — COMMENTED** [review]
generally lgtm, just some suggestions to consider

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 3 above)
Comment (zi-yii) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:155]: I thought it was a new word that I did not know of 😆

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 4 above)
Comment (yanqiyqh) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:158]: yup I agree but I think its easier to see as Claim Amount in camelcase

File: src/main/java/seedu/address/logic/commands/AddClaimCommand.java (hunk 2 above)
Comment (yanqiyqh) [reply src/main/java/seedu/address/logic/commands/AddClaimCommand.java:27]: I agree, addressing this in PR #96 as well, in the parser, we should allow for input in the form of 30.15 (as in $30.15) and convert it to integer 3015. It's more intuitive for the user to enter the amount that way.

**Review by yanqiyqh — COMMENTED** [review]
Overall LGTM, but just a few tweaks to make it more user-friendly.

File: src/main/java/seedu/address/logic/Messages.java (hunk 1 above)
Comment (RezwanAhmed123) [reply src/main/java/seedu/address/logic/Messages.java:59]: done!

File: src/main/java/seedu/address/logic/commands/AddClaimCommand.java (hunk 2 above)
Comment (RezwanAhmed123) [reply src/main/java/seedu/address/logic/commands/AddClaimCommand.java:27]: will fix according to suggestions below

Comment (RezwanAhmed123) [reply src/main/java/seedu/address/logic/commands/AddClaimCommand.java:27]: fixed in latest PR, will be linked later.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 3 above)
Comment (RezwanAhmed123) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:155]: no i was trolling LOL, pls ignore

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 4 above)
Comment (RezwanAhmed123) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:158]: i updated all the messages later on, you can view in the latest PR.

**Review by RezwanAhmed123 — COMMENTED** [review]
currently fixing all the nits from the PRs.

**Comment by RezwanAhmed123** [comment]
refer to PR #101

**Pull request closed without merging**